tests/storage/merge_test.py

```diff
@@ -522,6 +522,7 @@ def test_reduce_chunked_internal(self):
                 ('reduce', (base_vol.optimal_size(),), {}),
             ]
 
+    @pytest.mark.xfail(reason='leafs never allocate more than optimal')
```

Comment:
Marking test as xfail because it does not work after the fix does not make sense.

If the tests does not work now change it to test the real behavior. I wonder why this test succeeded before. We should test here real volume size (e.g. 10g).

The fix should not affect merge tests, it it breaks them the tests probably not testing real use case.

Reply:
That's why I removed it in the first place, xfail just shows that it is indeed failing in the pipelines.

It was working because we were overallocating for small volumes and the test levarages that. Bigger volumes (e.g., 10g) will not work, with current code or previous code.

In general what is making this test fail (vs other similar test) is that leaf volumes have an increased optimal_size, which makes it consistently bigger than the allocated size: 
https://github.com/aesteve-rh/vdsm/blob/aesteve/small-volumes-blocksd/lib/vdsm/storage/blockVolume.py#L432

Reply:
I'll close this thread to not to diverge from the other [thread](https://github.com/oVirt/vdsm/pull/293#discussion_r938228290), which is a related discussion (aka what to do with the test that now fails no matter which size we use).